### ISSUE TITLE
image added through camera in FM is blank until one leaves the page and comes back

### ISSUE THREAD

**Issue by lbestock** [urapadmin/kiosk#1151]
A new bug introduced somehow. I have now seen it in S'Urachi and URAP 10.7.1 but in both cases when I was showing things to someone and not able to stop to document. And now I can't replicate it. Both times on the iPad - I think once in dayplans and once in S'Urachi's ceramics module. I made a photograph using the camera, and then there was just a white blank where the photo should have been. When I exited the tab and reentered the photo appeared properly, so it was only after entry that it was weirdly blank.

I know this cannot count as a real ticket. I will try to run into the problem again tomorrow.

**Comment by lbestock** [comment]
Must be related and also super weird. I just logged in to S'Urachi to test this. And the start page looked normal for a second, and then the site map flashed off, to blank. So I went into the site, clicked on the site map, and there, too, it looked normal for a second and then flashed off. And then I went back to the home page and the map had reappeared. But then it flashed off AGAIN. So I went back to site again and then it looked normal, and this time it seems to have stayed. But there is something interfering with the display of images and it isn't consistent.

**Comment by lbestock** [comment]
Now seen again in context photos. First I made a dayplan with the camera and it worked just fine. Then I went to a preexisting context, went to the photos tab, added a new image container, took it with the camera, got the short flash that I expect when it is putting an image in, and then I get a blank screen.
![image](https://user-images.githubusercontent.com/49291909/122130085-19ec3180-ce05-11eb-8c3b-fbd4bf282b2c.png)

One can tell there really is an image there because I don't get the little "insert an image here" text.
Ah, more. Then I created another one. And that one came up pre-filled blank, without the "insert an image here" text! And then I tried to scroll down to make sure it really had added an image, and indeed there were two apparently blank image containers....for a moment. And then one of them disappeared. And then I added yet another image container, and for a moment there were three and now there are two blank ones (which, however, display as different sizes) and the third is missing but there is a big blank grey area and no description field where it should be. Wow this is crazy. Then I logged out and logged in again and went back to the photos for that context. And on first scroll, two blank ones with the "insert image here" text, and then at the bottom the image I had actually taken with the camera, as it should be. And then I scroll up, and then I scroll down again, and for a moment I can't get anywhere at all. But then I am able to scroll again and I have two blank containers (no text) and my image. 

In sum: image containers have gotten messed up, apparently across the board, on the displaying of images. While I run into related problems repeatedly, the same steps do not always result in the same problem (I just successfully made and can see an apparently stable photo in a preexisting context). That things seem unstable, so the condition of what I can view changes without me touching anything, seems particularly weird to me and I do wish I had not discovered that while you were in the midst of moving, so I'll go back to quilting.

**Comment by urapadmin** [comment]
- do you have the same thing in any other project but SU?
- does it happen on your ipad?
-does it happen on a different mac?
-

**Comment by lbestock** [comment]
I'll test more thoroughly tonight but it has so far only been encountered on iPads - both mine and the one I gave to Peter - and has been seen in SU and URAP.

**Comment by urapadmin** [comment]
that sounds really bad. Which FM Go Version?

**Comment by lbestock** [comment]
19. It's bad enough I would try to get Peter to download again once you've fixed it.

**Comment by urapadmin** [comment]
that sounds clearly like a fm bug. Just fixing it sounds unlikely.

**Comment by urapadmin** [comment]
this sounds like the flickering issue I had to give up on: #791. I need the exact versions and projects where it does occur and I need to know where it does not occur. I need to trace back the steps that might have interfered with ALL image containers.

**Comment by lbestock** [comment]
So far I am only certain of it in SU 10.7.1 (that is still called 10.7 but we know that is wrong). It has been seen on my iPad and the black iPad, both of which are running FM Go 19. On mine I can check: 19.2.2, but the black iPad is in the Netherlands. However, it should be pretty up to date because I had to install FM Go 19 just before Peter left.

I thought, when I ran into the problem with Peter, that I remembered running into it in URAP shortly before. But by the time I thought to test it I had deleted my last URAP download and so I am not actually sure - I may have first hit it when testing a previous iteration of SU, and since URAP should be a version behind ARCH 1900 (according to the release history) and they say for sure they are not running into this it seems unlikely. I will check URAP once my tea has steeped to put my mind at ease, and I can do USTP after that to check that 10.6.3 really is clean.

If you want me to check to see if it was between 10.6.3 and 10.7, or 10.7 and 10.7.1, you might have to get me a 10.7 - no one is currently running it.

**Comment by urapadmin** [comment]
has your tea steeped?

**Comment by lbestock** [comment]
Twice over and I had a crazy Natalies from Seven Stars. But I couldn't take care of this because of #1153. Now you've given me something to try there and I will get some more tea going...

**Comment by urapadmin** [comment]
unsurprisingly I cannot reproduce this easily (so far not at all). 
I downloaded your workstation from urap-test (Laurel's iPad) and added images to collected material and context photos. No issues (except that it is getting all a bit slow on the small yellow mini). 
I am on FM Go 19.0.1.88, iOs 14.1 and recording 10.7.1.

If you have a database where you can still reproduce it, please upload that exact database to urap-test. I will try if I can reproduce the issue with that same one.

**Comment by lbestock** [comment]
Yup, wow, still happening. I just added a photo of the peonies I need to get rid of, and then ALL the image containers went blank.
![image](https://user-images.githubusercontent.com/49291909/122432275-2eded700-cf63-11eb-9e1b-aeeaf2bf8517.png)
I have uploaded it and let you do the import and whatever else you want with it.

**Comment by lbestock** [comment]
I have this problem in URAP, too, 10.6.3, iOs 14.4.2, FM Go 19.2.2. So it has to be something in the operating system or FM version I have, right? Only it's also screwed up on Peter's which is the thing that worries me most right now. I've got a software update for the iOs ready and I will run that and see if it makes a difference...

**Comment by urapadmin** [comment]
I downloaded your version. 
- it does not happen with that database on FM 19.2.2 for windows.
- I cannot see it on the yellow ipad (still FM Go 19.0.1.88, iOs 14.1 and recording 10.7.1).

What about your mac? (You can download that database by downloading the x1lk workstation).

**Comment by urapadmin** [comment]
we would now need the versions from Peter's iPad since it happened there, too.

**Comment by lbestock** [comment]
I will try these things now. BUT after updating my iOs and going back into the file that was just causing me such problems, I cannot repeat this.

**Comment by urapadmin** [comment]
great

**Comment by lbestock** [comment]
No problem with the S'Urachi file on my Mac.

**Comment by urapadmin** [comment]
so it is gone after an update to iOS 14.6 in both URAP and s'Urachi?

**Comment by lbestock** [comment]
Checking S'Urachi now. There are a lot of pictures of Omar all of a sudden.

**Comment by urapadmin** [comment]
I don't have Omar. Was just photographing Ulli's desk. White. Nothing on it.

**Comment by lbestock** [comment]
How could you tell, then, if the error occurred, since it just came back white when it happened?

**Comment by urapadmin** [comment]
one of my cables is on the desk

**Comment by lbestock** [comment]
Many indistinguishable Omars later (she's pretty inert) I cannot reproduce this in either of the files that were most recently giving me trouble. We can check Peter's - it's the workstation called Peter. But there seems no point. What we need is for Peter to update the iPad.

**Comment by lbestock** [comment]
It was definitely in URAP the first time I saw it, now I remember: I was zooming with the publishing people. So there really can be no question: it's not the our stuff at all, just dumb Apple crap.

**Comment by urapadmin** [comment]
Another indication that the filemaker solution is not stable and often enough beyond my control. Has always worried me. 
Tell Peter to upgrade the iOS while in the Netherlands. And then we'll see if it helps. He is, of course, invited to join us all in Github.

**Comment by lbestock** [comment]
Um. Do you know the passcode for the black ipad? The wiki says urap but is that (8727) right? Because Peter is going to need it once he updates...

**Comment by urapadmin** [comment]
I don‘t know

**Comment by urapadmin** [comment]
Aren't you confusing the numerical code to get into the device with the password of the apple-id it is connected to. He could just use the apple-Id and password that is in the wiki for the other urap ipad. And you should update that wiki page with the last information and not with a history of changes, I think. I can't tell from the current information what is what.

**Comment by lbestock** [comment]
What he'll need is the numerical code to get into the iPad after it restarts. I have a hunch it's 0000 but I am not sure, and you are of course completely right about the wiki...

**Comment by lbestock** [comment]
Bloody hell I have run into this old nemesis again while testing pvd.
![CCF2F497-FCBD-4F68-861B-9867C0882FAE](https://user-images.githubusercontent.com/49291909/126879173-4a1f3664-19de-48c5-b6fc-cff83262d45e.png)
10.6.3

**Comment by lbestock** [comment]
FM go 19.3.1.
iOs 14.6.

**Comment by urapadmin** [comment]
does it happen repeatedly after closing fm go? Does it happen anywhere else? Can you provoke it?

**Comment by lbestock** [comment]
It happened four times, three in narrative and once CM. I have closed and reopened the file and now cannot provoke it again.

**Comment by urapadmin** [comment]
I thought so. Our only change is to find out what provoked in the first place. My feeling is that after it happens once one should close filemaker immediately and restart. This is not the only phenomenon that appears every now and then but never stays. Apparently arch1900, USTP and URAP don't encounter any such things.

**Comment by lbestock** [comment]
Aaand again in a new download newly opened.

**Comment by urapadmin** [comment]
counting really does not help

**Comment by lbestock** [comment]
Despite counting not helping I will add another instance. I downloaded and opened a new PVD (11.1). I made a new context and made images. I could not see my photo. I made a second image record and added a photo to it. But I couldn't even see that I had multiple image containers. I logged out and logged back in again and now all is fine, two photos and two empty image containers from me being impatient and clicking the + button. And I can add images now like I should be able to, so there's a lot of Omie being added to PVD.

**Comment by urapadmin** [comment]
it really does not help as long as you aren‘t trying to be systematic about it and find patterns. That it is a looming issue we know. Hence I have not equipped arch1900 with a new version. 
But you must at least record the fundamentals: ipad/ios version/filemaker version or this is without any use.

**Comment by urapadmin** [comment]
you could download arch1900 (you have to enable the workstation first) and try it there.

**Comment by lbestock** [comment]
LDB iPad (mini 5th generation)
iOS 14.7.1
Filemaker Go 19.3.1

I just downloaded arch1900 from the arch0100 kiosk and ran into this again. That is recording system v. 10.5.1.
I went to an existing locus that had no photos yet.
I made a new image container. 
Without giving it a description, I clicked in the image container and asked it to use the camera.
It flashes as it processes the image, but the result is a big blank.
On logging out and re-entering, I can see my image as it should be.
I created a new image container. Same thing - flashes as processing but I can't see the image.

And then it gets particularly weird. When I first tried scrolling through images to see if I could see my original image, I couldn't - only the blank image container. And then I put my iPad down to start typing this, and then when I picked it up again (I had not turned it off or logged out or anything, it was just sitting beside me with Filemaker open) I could scroll and see both the original image and the new blank. And so then I put it down again, similarly. And now when I pick it up I can only see the original, I cannot scroll to the new blank. It messes with my head, this one.

**Comment by urapadmin** [comment]
especially since this is the version being used in the arch1900 field and we have not heard about the issue from them. I really have no clue, currently.

**Comment by urapadmin** [comment]
@luizaogs: this is the second phenomenon that only @lbestock seems to encounter and nobody else.

**Comment by urapadmin** [comment]
@aes1884: Please have an eye on this, Anna and tell us if it occurs in class.

**Comment by luizaogs** [comment]
Ran into this on my first try creating a dayplan. Blank screen after photo was taken, had to quit FM but the photo was there when I entered again (it did ask me to log in). Happened three times in a row, all in dayplans, and then I stopped trying. This is Filemaker Go 19.3.1.22, iOS 14.4.2, PVD 11.5.

Though wait, I just realized that this is the problematic iOS version from above and I can't just update my friend's iPad though she clearly needs an update. Hmpf. Well then this is useless.

**Comment by urapadmin** [comment]
I am trying to reproduce this on my own ipad and fail in dayplans. So I need more detailed steps.
First: If you install FM GO 19 you DO NOT uninstall FM18 or any other FM Go version installed. I get trapped by that over and over again. So please make sure that you delete all filemaker go versions except the current (which is the first with the Claris icon). I will add  a version info in the next database so that it is easier to actually see what version one is in.

**Comment by urapadmin** [comment]
there is a new recording version now that shows the system information on the start page

**Comment by lbestock** [comment]
FM Go 19 is the only version on my iPad.
The new version shows me I am in 19.3.1.
I have gone into Playground unit FL.
I created a dayplan, clicked in the "insert image here", and chose camera.
I take a picture. I get one flicker, and then the keyboard pops up on an otherwise blank screen. This IS different than this problem as first encountered, where one had a blank image container. Now the blankness is total: 
![image](https://user-images.githubusercontent.com/49291909/139751314-2517c394-f68b-475d-871b-d186262d0b7e.png)

I hadn't checked the filepicking rules but will do so now.

**Comment by luizaogs** [comment]
Testing PVD in PVD, how strange.

Filemaker 19.3.1.23 (the only one in the iPad now), iOS 15.0.2. I went straight to unit LA, then straight to the dayplans tab. Then I immediately clicked on the plus icon and then on the image container to insert the image. Then clicked on camera,  took the picture, clicked on use photo, got your favorite blank screen in which the keyboard was also open. Then clicked on Files on the top left (accidentally) and went to Files, and then went back to FM and the image was NOT there. Hm. 

@lbestock since we're evidently testing the same thing at the same time (ha), is your image there when you quit FM and go back?

**Comment by lbestock** [comment]
@luizaogs Yes, it is! Testing in PVD making problems in Bonn....

**Comment by luizaogs** [comment]
Weird. Now I tried to see if I could do it in a way in which my image would be there once I got back and I DIDN'T get a blank screen, it went straight back to the dayplans screen, but there's no image added (I did the same thing to add the dayplan as above, only this is the same database as before so not a fresh download). But then I tried again and got the blank, pressed the home button twice and went back to FM and the image was there. There's a file picking rule making images in LA low resolution.

Well I suspect this is unhelpful and I'm confusing myself so I'm going to await further testing instructions :)

**Comment by urapadmin** [comment]
can you confirm that it happens only once after a fresh download and when you close and reopen filemaker it does not happen again with that same file? 
We really need to find a pattern here. I cannot reproduce it at all here with the yellow ipad (old but most recent software versions)

**Comment by urapadmin** [comment]
Please test in general when you encounter it: 
Once the screen is white blank does switching to a different app (double home screen tap) and back change the situation or does file maker stay the same?

**Comment by urapadmin** [comment]
hmpf hmpf hmpf.

**Comment by lbestock** [comment]
I am just going to try things and report them here. 
I had quit FM go (double tap, swipe up to close). I went back to FM and it asked me to log in again. I did, went back to FL, and made a new dayplan. I went to camera. This time I got a double flicker, and then I got not a blank screen, just a blank image container. It also blanked out the previous image containers. 
![image](https://user-images.githubusercontent.com/49291909/139769457-abbc2930-1cff-4534-b4bb-68a28017f58b.png)
In this case I did not quit FM. I simply went back to the home page, and then re-entered FL. And there were my images:
![image](https://user-images.githubusercontent.com/49291909/139769529-41ff38dd-f636-4145-9fc4-879c2f773939.png)

**Comment by luizaogs** [comment]
> can you confirm that it happens only once after a fresh download and when you close and reopen filemaker it does not happen again with that same file?

I just got the same issue as in Laurel's comment above (blank container rather than screen, all other images disappearing, etc.). But what I was trying to see was whether this happened again once I quit FM and returned to the same database and, alas, yes. I logged out, quit FM and returned, and now I got the full blank screen rather than just the empty image container when trying to add another dayplan.

> Please test in general when you encounter it: Once the screen is white blank does switching to a different app (double home screen tap) and back change the situation or does file maker stay the same?

I then switched to a different app with the double home screen tap and went back to FM, and at that point I got the dayplans screen with my image on it.

**Comment by urapadmin** [comment]
### #662 
okay, no way to reproduce this on the yellow ipad with exactly the same iOS and FM Go versions. So this is a much older ipad than yours and perhaps it is a timing issue? Since you can reproduce this rather reliably you have to be my guinea-pigs:

There is a new recording version 11.5-test installed ob pvd.
Under registrars and specialists you find this now:
![image](https://user-images.githubusercontent.com/38838314/139954573-e09331cd-2b8c-41c1-a857-22e60a31c71a.png)

- [ ] Please try twice if you still have the trouble in dayplans - start fm go afresh before you try the second time.
- [ ] if you can reconstruct the issue twice, restart the database again and switch on test_1 - try again - what happens? 
- [ ] leave, start again, switch off test_1 and try again if you still get the issue
- [ ] leave, start again, switch on test_2 and try again. What happens?

**Comment by luizaogs** [comment]
Had the issue twice when adding dayplans to LA (both times had the total blank screen with keyboard). My images were not there when I returned after closing the app and logging in again.

Restarted the database again, turned on test 1. Went to LA again, added a dayplan. Blank screen with keyboard popped up again.

Left, started again, but test 1 had turned off automatically? So I didn't have to turn it off. Anyway, tried adding a dayplan then and noticed that my image from the latest dayplan was now there (the one added when test 1 was turned on). The two earlier ones are nowhere to be found. Got the total blank screen with keyboard again.

Left, started again, turned on test 2. My image was also back there this time (so the dayplan I added after test 1 had been turned off). With test 2 on, I tried adding another dayplan. This time it worked! No issues, image added to image container, cursor went to the description box. Oh wait, dammit -- then I clicked outside of the description box and the image disappeared: 

![IMG_F3F26740EE51-1](https://user-images.githubusercontent.com/67970402/139957064-bcc6c9a2-0ed3-4b77-9513-92995391d434.jpeg)

But then I clicked outside of the dayplan and it was back, so I don't know what that was:

![IMG_F53A0463F107-1](https://user-images.githubusercontent.com/67970402/139957162-d9301bc9-399d-4dd3-8322-59489579b77b.jpeg)

(Is this all clear? If not let me know.)

**Comment by urapadmin** [comment]
yes, quite clear. Also quite unfortunate. Here comes more:

This time please make sure test mode is on:
![image](https://user-images.githubusercontent.com/38838314/139958642-3b4c16ce-23d8-4eac-8948-14283e8116c1.png)

- Then activate dayplan_test_1 and do the awful thing. 
- When you get the blank screen please try switching to a different app and back, 
- hoping that when you turn back to filemaker the blank screen is gone.
- If that is so please click on the test-log button and copy the contents to your clipboard and then paste it here.

**Comment by urapadmin** [comment]
btw: Yes, it is correct, the dayplan_test checkboxes are reset when you close filemaker.

**Comment by luizaogs** [comment]
Alright, test mode on, dayplan test 1 on. Got the blank screen as you expected. Switched to different app and back, blank screen was still there. Hmm. Now what? 

Oh wait, if I minimize the keyboard, then double tap, then come back it takes me back to dayplans. Test log incoming ...

**Comment by luizaogs** [comment]
switch_off_menu: Switched the menu off. isMasterDb = yup
switch_off_menu: Switched the menu off. isMasterDb = nope
close_temp_window  temp_window was not open to begin with.
switch_off_menu: Switched the menu off. isMasterDb = nope
switch_off_menu: Switched the menu off. isMasterDb = nope
activate_temp_window: opened layout dayplan in NEW window  temp_window on iOS
closed temp window  temp_window.
save_image: enter for img
save_image: enter for img
add_dayplan_to_cache: single image with param OnObjectModify, $$cache_done=
add_dayplan_to_cache: image missing: 0/1
is_cache_update_needed: image is there in OnObjectModify but record is not-> DONT process cache
add_dayplan_to_cache: doing nothing.
image_validate: field img
add_dayplan_to_cache: single image with param , $$cache_done=0
add_dayplan_to_cache: image missing: 0/0
is_cache_update_needed: OnValidate but NOT CACHE_DONE-> process cache
activate_temp_window: opened layout file_identifier_cache in NEW window  temp_window on iOS
closed temp window  temp_window.
add_dayplan_to_cache: deleted cache entries.
activate_temp_window: opened layout file_identifier_cache in NEW window  temp_window on iOS
closed temp window  temp_window.
add_dayplan_to_cache: installing timed refresh.

**Comment by urapadmin** [comment]
> 
> Oh wait, if I minimize the keyboard, then double tap, then come back it takes me back to dayplans. Test log incoming ...
Interesting. Can you please try a run with the floating keyboard? Not sure if you know, you switch to a different keyboard type by tapping on the keyboard button in the lower left corner for longer. Then a menu appears and you get access to different keyboard types. I would be interested in a run with the floating keyboard. But switch to it BEFORE you do the photo thing.

**Comment by urapadmin** [comment]
thanks for the log. Looks totally normal, of course. Hmpf.

**Comment by luizaogs** [comment]
With the test mode on or not?

**Comment by urapadmin** [comment]
no, not necessary.

**Comment by luizaogs** [comment]
Ok, I downloaded a fresh database. Floating keyboard is on.

Tried adding a dayplan to LA, blank screen with floating keyboard. Quit FM and came back, my image wasn't there. Tried to add another dayplan, blank screen with floating keyboard.

Quit FM, returned, turned dayplan test 1 on. My previous two images were not there. Tried adding a dayplan with test 1 on, blank screen with floating keyboard.

Quit FM, returned, dayplan test 1 was off. The image I had added while using test 1 was where it should be. Tried adding another dayplan, blank screen with floating keyboard.

Quit FM, returned, turned on dayplan test 2. This time the image I had taken while test 1 was off was not there. Tried adding another dayplan. This time it didn't work, but I didn't get a blank screen and the cursor went straight to the description box: 

![IMG_905ED17D3E83-1](https://user-images.githubusercontent.com/67970402/139961902-d97731ff-2ad4-4fe0-81e3-5773b63d18b6.jpeg)

Going back to home and re-entering LA didn't show my image, and it also wasn't there once I quit FM and logged back in.

**Comment by urapadmin** [comment]
this is so messed up :( You weren‘t even successful while dayplan_text_2 was switched on? That is really bad.

**Comment by luizaogs** [comment]
Not this time, but it worked the first time so I'm not sure. I just turned on the regular keyboard again, though, and that worked: 

![IMG_7CD69C81A867-1](https://user-images.githubusercontent.com/67970402/139963341-e64d9401-3540-476f-8999-d78daaa9d1c7.jpeg)

**Comment by luizaogs** [comment]
Ok, tried floating again and that definitely does not work.

**Comment by urapadmin** [comment]
a mess. Thank you. Currently I am out of ideas.

**Comment by luizaogs** [comment]
Argh I'm sorry!

**Comment by urapadmin** [comment]
one last thing. Can you try if inserting a scrible with the signature option instead of the camera makes a difference (test modes off, normal keyboard).

**Comment by luizaogs** [comment]
Hmm then I didn’t get the blank screen and the cursor went to the description box but the sketch didn’t show up in the image container, not even after I left and came back.

**Comment by luizaogs** [comment]
Wait, now it worked. I didn’t do anything differently. Weird…

**Comment by urapadmin** [comment]
ja, weird. Welcome to Filemaker. I am clueless.

**Comment by lbestock** [comment]
I've got the purple iPad in my hot little hands (@Tfranconi offered to help test but our teaching schedules and a hiccup meant that I am the one with the ipad at the end of the day).
PVD v11.5-test running on iOS 15.1 (iPad), Filemaker Go iPad 19.3.1.23.
Tyler updated it all.
I loaded up a new workstation for PVD.
Dayplan, no problem. I note: it flickered more than it does before it fails on my iPad.
Context photo, no problem.
Locus relations photo: crash.
Exit, re-enter, locus relation with a signature sketch, no problem.
Without going anywhere, make another locus relation. Photo. Fine. (The "actions" dialog box that starts with view, replace... pops up automatically after I make a locus relation image, whether it is a photo that crashes or a signature or photo that is successful.)
This all strikes me as much more what it was like months ago when it was a bit annoying but not totally malignant yet. I fear it will not help you understand much.

**Comment by lbestock** [comment]
Here is a test log from me after a blank screen locus relation (which itself came after successful dayplan and context images). No more from Purple until instructions and sleep and nine zillion DGS things.

switch_off_menu: Switched the menu off. isMasterDb = yup
switch_off_menu: Switched the menu on. isMasterDb = yup
switch_off_menu: Switched the menu off. isMasterDb = yup
activate_temp_window: opened layout dayplan in NEW window  temp_window on iOS
closed temp window  temp_window.
save_image: enter for img
save_image: enter for img
add_dayplan_to_cache: single image with param OnObjectModify, $$cache_done=
add_dayplan_to_cache: image missing: 0/1
is_cache_update_needed: image is there in OnObjectModify but record is not-> DONT process cache
add_dayplan_to_cache: doing nothing.
image_validate: field img
add_dayplan_to_cache: single image with param , $$cache_done=0
add_dayplan_to_cache: image missing: 0/0
is_cache_update_needed: OnValidate but NOT CACHE_DONE-> process cache
activate_temp_window: opened layout file_identifier_cache in NEW window  temp_window on iOS
closed temp window  temp_window.
add_dayplan_to_cache: deleted cache entries.
activate_temp_window: opened layout file_identifier_cache in NEW window  temp_window on iOS
closed temp window  temp_window.
add_dayplan_to_cache: installing timed refresh.
add_dayplan_to_cache: single image with param , $$cache_done=0
add_dayplan_to_cache: image missing: 0/0
is_cache_update_needed: OnValidate but NOT CACHE_DONE-> process cache
activate_temp_window: opened layout file_identifier_cache in NEW window  temp_window on iOS
closed temp window  temp_window.
add_dayplan_to_cache: deleted cache entries.
activate_temp_window: opened layout file_identifier_cache in NEW window  temp_window on iOS
closed temp window  temp_window.
add_dayplan_to_cache: installing timed refresh.
save_image: enter for img
save_image: enter for img
add_locus_photo_to_cache: single image with param OnObjectModify, $$cache_done=0
add_locus_to_cache: image missing: 0/1
is_cache_update_needed: image is there in OnObjectModify but record is not-> DONT process cache
add_locus_photo_to_cache: doing nothing.
image_validate: field img
add_locus_photo_to_cache: single image with param , $$cache_done=0
add_locus_to_cache: image missing: 0/0
is_cache_update_needed: OnValidate but NOT CACHE_DONE-> process cache
activate_temp_window: opened layout file_identifier_cache in NEW window  temp_window on iOS
closed temp window  temp_window.
activate_temp_window: opened layout file_identifier_cache in NEW window  temp_window on iOS
closed temp window  temp_window.
activate_temp_window: opened layout file_identifier_cache in NEW window  temp_window on iOS
closed temp window  temp_window.
add_locus_photo_to_cache: single image with param , $$cache_done=0
add_locus_to_cache: image missing: 0/0
is_cache_update_needed: OnValidate but NOT CACHE_DONE-> process cache
activate_temp_window: opened layout file_identifier_cache in NEW window  temp_window on iOS
closed temp window  temp_window.
activate_temp_window: opened layout file_identifier_cache in NEW window  temp_window on iOS
closed temp window  temp_window.
activate_temp_window: opened layout file_identifier_cache in NEW window  temp_window on iOS
closed temp window  temp_window.
switch_off_menu: Switched the menu off. isMasterDb = yup
add_locus_relation_to_cache: single image with param , $$cache_done=0
is_cache_update_needed: OnValidate but NOT CACHE_DONE-> process cache
add_locus_relation_to_cache: no opposite relation found for locus relation 0359F3F8-61EC-4EA3-9BDE-600ECC0882C5
activate_temp_window: opened layout file_identifier_cache in NEW window  temp_window on iOS
closed temp window  temp_window.
add_locus_relation_to_cache: row_nr is  relation uid is 0359F3F8-61EC-4EA3-9BDE-600ECC0882C5
activate_temp_window: opened layout locus_relations in NEW window  temp_window on iOS
closed temp window  temp_window.
save_image: enter for img
current_locus_relations_on_modify_image (start): Sketch missing: img missing=0/uid_sketch missing=1
locus_uid: d5899905-32a1-4af2-a8e0-df603fe37c76/locus_related_uid=97467ece-10b6-4bed-996c-3fa6ae144595
activate_temp_window: opened layout locus_relations in NEW window  temp_window on iOS
closed temp window  temp_window.
current_locus_relations_on_modify_image: locus relation 0359F3F8-61EC-4EA3-9BDE-600ECC0882C5 got sketch 
current_locus_relations_on_modify_image: calling add_locus_relation_to_cache
current_locus_relations_on_modify_image (<>dont_add): Sketch missing: img missing=0/uid_sketch missing=1
add_locus_relation_to_cache: single image with param OnObjectModify, $$cache_done=0
is_cache_update_needed: image is there in OnObjectModify but record is not-> DONT process cache
add_locus_relation_to_cache: doing nothing.
current_locus_relations_on_modify_image (after sub call): Sketch missing: img missing=0/uid_sketch missing=1
image_validate: field img
add_locus_relation_to_cache: single image with param , $$cache_done=0
is_cache_update_needed: OnValidate but NOT CACHE_DONE-> process cache
activate_temp_window: opened layout file_identifier_cache in NEW window  temp_window on iOS
closed temp window  temp_window.
add_locus_relation_to_cache: row_nr is  relation uid is 0359F3F8-61EC-4EA3-9BDE-600ECC0882C5
add_locus_relation_to_cache: before sub call row_nr is 1 relation uid is 0359F3F8-61EC-4EA3-9BDE-600ECC0882C5
current_locus_relations_on_modify_image (start): Sketch missing: img missing=0/uid_sketch missing=0
locus_uid: d5899905-32a1-4af2-a8e0-df603fe37c76/locus_related_uid=97467ece-10b6-4bed-996c-3fa6ae144595
activate_temp_window: opened layout locus_relations in NEW window  temp_window on iOS
closed temp window  temp_window.
current_locus_relations_on_modify_image: locus relation 0359F3F8-61EC-4EA3-9BDE-600ECC0882C5 got sketch 31F3BF5A-F3CB-469B-9CD6-14D23014DEB1
add_locus_photo_to_cache: on_modify_image for opposite relation returned 1
add_locus_relation_to_cache: after sub call row_nr is 1 relation uid is 0359F3F8-61EC-4EA3-9BDE-600ECC0882C5
save_image: enter for img
save_image: saving image data for field

**Comment by urapadmin** [comment]
We must focus here. Please stick to dayplans and don't go to locus relations for the time being. 
So no matter how often you try dayplans on that purple ipad, you do not get the issue, is that correct?

**Comment by urapadmin** [comment]
can somebody please send me a screenshot from one of the new ipad's camera settings? Just go to settings, Camera and give me a screenshot of the options there. I just wonder if Apple has added anything to the camera that causes this effect in filemaker. As they once did.

**Comment by urapadmin** [comment]
> 
> 
> Wait, now it worked. I didn’t do anything differently. Weird…
> 
> That worked twice. Then I quit and came back and tried adding a new one and got a blank container.

you got the blank container adding a scribble?

**Comment by urapadmin** [comment]
@luizaogs: I have made you an arch1900 user logs. You will have to log in first with an empty password and define one.
Then you should have access to the permanent download workstation. Can you please download it and see if you can provoke the same trouble in arch1900 dayplans? 

@lbestock: You may do the same thing with the ipad that has most frequently produced the issue.

**Comment by urapadmin** [comment]
Different test (please be disciplined and try not to trip yourself up with all the different tests): 

You find a file in the PVD file repository if you use the tag "horrors". It is an empty pvd database. Please download it (you will have to log in as admin since it is really totally empty. Doesn't even have most of the configuration settings!). 
Just add a site and a unit and try to provoke the error. The idea here is to see if it has anything to do with a mass of data or happens even in an empty database.

**Comment by urapadmin** [comment]
> 
> 
> Ok, tried floating again and that definitely does not work. (This and the one above are with dayplan test 2 on.)

I tried the floating keyboard on my older ipad. Works perfectly fine. What is it about these new ipads that is different?

**Comment by lbestock** [comment]
I hit it in dayplans on the Purple ipad. 
I went to a unit from the start page, entered dayplans, made a new one, it was fine.
I went back to start, chose a different unit, entered dayplans, made a new one, crash.

**Comment by lbestock** [comment]
First try test with arch1900 was the original version of the problem: blank image container but not frozen out of FM.

**Comment by lbestock** [comment]
I repeated my steps exactly on the Purple ipad (the above test was on my own, gen 5 iPad mini). I even went to the same unit to make my first dayplan. And it worked.

Oh argh argh triple argh. It worked. And then back to the super weirdness. I repeated my next step, which was going to a different unit and making a new dayplan. It seemed to work, but then I noticed that the narratives were looking weird and grey, as per a ticket we just closed, #1187. And I tried to scroll a bit, and it was not responsive, and then it jumped to a screen where the narratives were partly grey and my most recent dayplan, which had seemed to work, is visible down below in the preview but does not show up in the list view. 

![9D23F3E0-493C-4A60-B1B4-6FBC8FC2FF86](https://user-images.githubusercontent.com/49291909/140428549-f487543a-51fd-4edf-84b8-b0fe1aef8556.jpeg)

**Comment by lbestock** [comment]
"Horrors" first try with an empty database: crash. And the total blankness kind, where one has to quit FM, not the blank image container kind. This was on iPad mini 5, my own.

**Comment by urapadmin** [comment]
can you please go through the dayplan_test_1 and _2 steps up above in your ipad with the empty db?

**Comment by lbestock** [comment]
Yes. Meanwhile I tried to log in to test PVD on the Purple iPad and my password isn't working?

**Comment by urapadmin** [comment]
sorry, you were in arch1900. Try the empty database first just without the dayplan_test switches

**Comment by urapadmin** [comment]
password issue not related.

**Comment by lbestock** [comment]
I did both those already. My first try with the empty one ("horrors") crashed without the dayplan_test switches.

**Comment by lbestock** [comment]
I did not have to log back in, but could copy the test log:
switch_off_menu: Switched the menu off. isMasterDb = yup
switch_off_menu: Switched the menu on. isMasterDb = yup
switch_off_menu: Switched the menu off. isMasterDb = yup
activate_temp_window: opened layout dayplan in NEW window  temp_window on iOS
closed temp window  temp_window.
save_image: enter for img
save_image: enter for img
add_dayplan_to_cache: single image with param OnObjectModify, $$cache_done=
add_dayplan_to_cache: image missing: 0/1
is_cache_update_needed: image is there in OnObjectModify but record is not-> DONT process cache
add_dayplan_to_cache: doing nothing.
image_validate: field img
add_dayplan_to_cache: single image with param , $$cache_done=0
add_dayplan_to_cache: image missing: 0/0
is_cache_update_needed: OnValidate but NOT CACHE_DONE-> process cache
activate_temp_window: opened layout file_identifier_cache in NEW window  temp_window on iOS
closed temp window  temp_window.
add_dayplan_to_cache: deleted cache entries.
activate_temp_window: opened layout file_identifier_cache in NEW window  temp_window on iOS
closed temp window  temp_window.
add_dayplan_to_cache: installing timed refresh.

**Comment by urapadmin** [comment]
what does that mean? It happened with the empty database with test_1 switched on?

**Comment by lbestock** [comment]
> what does that mean? It happened with the empty database with test_1 switched on?

Yes, that's what I mean.

**Comment by urapadmin** [comment]
okay. Now test_2

**Comment by lbestock** [comment]
Then I did it with test_2 switched on. This time it worked, sort of...
The image container below has the image. The image does NOT show up in the list. I copied the log but hang on that's a different device...

**Comment by urapadmin** [comment]
I don‘t need the log

**Comment by lbestock** [comment]
Screen shot? I got that, too.

**Comment by urapadmin** [comment]
no. So new ipad, empty db works with test_2 switched on. Please repeat it a few times to make aure it ALWAYS works in that combination

**Comment by lbestock** [comment]
But it doesn't entirely work. It should show up in the list immediately. That doesn't worry you?

**Comment by urapadmin** [comment]
no

**Comment by lbestock** [comment]
So far good. I cannot make a different type of unit without going into configuration, but in two excavation units I have not been able to provoke this in dayplans, whether I go there straight after start or come from somewhere else. Most notably it never varies in its behavior, same amount (really none) flashing, it doesn't screw up the narrative, it doesn't ever appear in the list and it never fails to appear in the container below, whether camera or signature scribble or existing photo.

**Comment by urapadmin** [comment]
ok

**Comment by urapadmin** [comment]
and when you leave dayplans and go back all the images should be in the list as well, right?

**Comment by lbestock** [comment]
Yes.

**Comment by luizaogs** [comment]
> > Wait, now it worked. I didn’t do anything differently. Weird…
> > That worked twice. Then I quit and came back and tried adding a new one and got a blank container.
> 
> you got the blank container adding a scribble?

Yes. Twice.

**Comment by luizaogs** [comment]
I’m slightly lost. @urapadmin do you need me to do the arch1900 and Horrors tests or do Laurel’s results above suffice? (I am happy to do them, just trying to avoid driving you insaner.)

**Comment by urapadmin** [comment]
no, thank you. I got enough from Laurel.

**Comment by urapadmin** [comment]
Last test:
there is a new fmp12 under tag "horrors" in the file repository. It is an empty pvd version from September 2020 when we did not have the issue. I want to test this to make sure that there is not some structural damage to the database template itself that causes this. That structural damage would not have been there 09/2020. So please just torture dayplans a bit until you hit the blank screen issue. As soon as that is the case you can stop and we know that our issue here is indeed a combination of new ipad / new iOs / new filemaker and nothing can be done about it with the current code.

**Comment by urapadmin** [comment]
oh, wait, there is a second test I want you to try: Try site map. That is the simplest image container we have. 
- try it with the current pvd database 
- try it with the one under "horrors"

Does it behave like dayplans? I mean, can you, at least once, produce the blank screen issue?
(It might need a bit more prodding, so don't give up too early.)

**Comment by luizaogs** [comment]
> Last test for either @lbestock or @luizaogs, depends on who is quicker:
> 
> There is a new fmp12 under tag "horrors" in the file repository. It is an empty pvd version from September 2020 when we did not have the issue. I want to test this to make sure that there is not some structural damage to the database template itself that causes this. That structural damage would not have been there 09/2020. So please just torture dayplans a bit until you hit the blank screen issue. As soon as that is the case you can stop and we know that our issue here is indeed a combination of new ipad / new iOs / new filemaker and nothing can be done about it with the current code.
> 
> Please use your newer ipads since they have been more reliable in producing the issue.

So far I have not seen the blank screen in several tries (I'll keep trying), but all of my images are blank containers and I don't get the pictures even if I log out of FM and come back, so still bad?

**Comment by luizaogs** [comment]
Site map in horrors, on the other hand, worked immediately with no blank containers (here is a cute cat): 

![IMG_74943F8BB836-1](https://user-images.githubusercontent.com/67970402/140572197-442e56a7-6cb9-4b02-b44b-dec4d46a8b4f.jpeg)

I'm going to try to provoke the blank screen.

**Comment by luizaogs** [comment]
My first attempt adding a site map for PVD, however, resulted in the blank screen. Only it was different than the dayplans blank screen: no keyboard and the actions popup instead: 

![IMG_1DE009BF0D1C-1](https://user-images.githubusercontent.com/67970402/140573178-6bea004c-4beb-4e5b-894e-334e7d302213.jpeg)

**Comment by urapadmin** [comment]
> 
> 
> > Last test for either @lbestock or @luizaogs, depends on who is quicker:
> > There is a new fmp12 under tag "horrors" in the file repository. It is an empty pvd version from September 2020 when we did not have the issue. I want to test this to make sure that there is not some structural damage to the database template itself that causes this. That structural damage would not have been there 09/2020. So please just torture dayplans a bit until you hit the blank screen issue. As soon as that is the case you can stop and we know that our issue here is indeed a combination of new ipad / new iOs / new filemaker and nothing can be done about it with the current code.
> > Please use your newer ipads since they have been more reliable in producing the issue.
> 
> So far I have not seen the blank screen in several tries (I'll keep trying), but all of my images are blank containers and I don't get the pictures even if I log out of FM and come back, so still bad? Well ok, one of them I got back upon logging in again but there really seems to be no rhyme or reason to it.
> 
> I have 12 dayplans. No blank screen, but all blank image containers.

I really despair about this. What dies blank image containers mean, are they still empty when you leave the unit and go back in or are they all there then?

**Comment by urapadmin** [comment]
only that cat belly keeps me from screaming

**Comment by luizaogs** [comment]
They're all blank. As I said one of the images came back after logging out of FM and returning, but the others were all blank upon adding the image and even after leaving and coming back. But it looks like the image was added because I get the usual flickers and the cursor goes to the description box.

**Comment by luizaogs** [comment]
> only that cat belly keeps me from screaming

She has deserted me, otherwise I would give you more.

**Comment by urapadmin** [comment]
I read you but I am just in this state of disbelief. Why the heck have we not seen that in 2020?

**Comment by urapadmin** [comment]
Luiza, can you please tell me what ipad generation you have?

**Comment by luizaogs** [comment]
6th

**Comment by urapadmin** [comment]
you had to login (file-maker style) in that database you got under "horrors"?

**Comment by urapadmin** [comment]
or better asked: You definitely find yourself in version 9.5.3 ?

**Comment by luizaogs** [comment]
The login was this screen: 

![IMG_0221](https://user-images.githubusercontent.com/67970402/140575461-5bffeb75-e024-4a92-9086-f719e308c45c.PNG)

But yes, the database itself tells me it's 9.5.3.

**Comment by urapadmin** [comment]
### recording 11.5-test-2
There is a new version on pvd now that tries an entirely different way to insert the dayplan. 
If you go to the Guinea Pig section you'll find a "dayplan_in_situ" checkbox now. If you use that one the new way of handling dayplans is being used:

- It should not flicker at all when you insert the image 
- BUT at the end there is an unfortunate switch of the current tab for a short time. 

More important: Do you get any side effects apart from the short tab switch?
Can you find your image in the gallery afterwards under the unit's id?

**Comment by luizaogs** [comment]
It worked! No side effects that I can see other than the tab switch, no flickering, I see my images in the gallery.

**Comment by lbestock** [comment]
That cat is fat. And so much hair on the belly. Who has heard of such a thing?

**Comment by luizaogs** [comment]
Ah but one can’t touch the belly. It’s forbidden.

**Comment by lbestock** [comment]
Second dayplan added this way resulted in blank image container and all other image containers in dayplans for that unit also looking blank. They are all there when I go out and in again, and they are properly in the gallery, but the blank problem reared its head again.

**Comment by urapadmin** [comment]
you see the "experimental" in the upper right corner?

**Comment by lbestock** [comment]
Yes.

**Comment by urapadmin** [comment]
so now Luiza's and your ipad behave differently. This is really not software engineering anymore. This is Topfschlagen.

**Comment by lbestock** [comment]
Luiza's is a different generation than mine, though.

**Comment by urapadmin** [comment]
but both ipads behaved similarly so far. With this take I have switched off what was supposed to be the problematic code entirely.

**Comment by lbestock** [comment]
They didn't entirely behave similarly. I have never run into the issue of a photo I had taken not being there after I exited, for instance, which Luiza did.

**Comment by lbestock** [comment]
I think. This has been a long ticket and my memory is questionable.

**Comment by urapadmin** [comment]
at least they reliably created problems both when one of them created problems. This here worked on Luiza's ipad (and I really expected it would). However, I can't see a pattern anymore at all. I just want to throw filemaker against a wall. And then trample on it. And then ...

**Comment by urapadmin** [comment]
so far I suppose you have always added a dayplan and then the image, right? Can you please try to just change images of existing dayplans? Does that have the same effect?

**Comment by lbestock** [comment]
That is correct. Trying now...

**Comment by urapadmin** [comment]
oh. And could you please close your ten thousand apps in the background? That could be a difference to Luiza. Perhaps it has something to do with low resources.

**Comment by lbestock** [comment]
OK I'll do that next. But also I can't provoke it at all right now, not by making new dayplans, not by replacing dayplans...

**Comment by urapadmin** [comment]
great.

**Comment by lbestock** [comment]
Uh. I haven't been able to provoke this again but now something else weird is going on. I made five new ones in three units with no problems. Then I started some more replacements. Having done a few I noticed something odd: two adjacent images got replaced with the same photo when I took a new photo. You can see this in the list: the highlighted one and the one below it both got this image, which I took only from the container of the highlighted one.
![image](https://user-images.githubusercontent.com/49291909/140622768-af441017-06b8-4cd8-9b39-941c73c3f6c9.png)

**Comment by urapadmin** [comment]
I suppose that could happen if those two had the same photos to begin with. E.g. from inserting it from the gallery. You have to try that again and in a less cluttered environment. I would call this a different issue for the time being.
That you do not get the blank error again is just a new escalation of horror. It means we know it is there lurking but we can't reliably reproduce it anymore.

**Comment by lbestock** [comment]
Which, perhaps it does not need saying, I saw twice and now cannot get to repeat.

**Comment by urapadmin** [comment]
how often have you gotten the blank error in this experimental mode?

**Comment by lbestock** [comment]
OK I'll ignore that one for the time being.
When we first saw this error, it seemed to happen on the first opening after download but not again reliably.
Let me see if I scrap this and download again what happens.

**Comment by lbestock** [comment]
Just once.

**Comment by urapadmin** [comment]
and you are absolutely sure that you were in experimental at that point and had just started filemaker?

**Comment by lbestock** [comment]
I am sure I was experimental and I am sure I had just opened the file for the first time and that no others were open.
![image](https://user-images.githubusercontent.com/49291909/140622867-50a51fc3-f3b2-4e5d-888b-54b2cf3c8949.png)

**Comment by urapadmin** [comment]
what is that?

**Comment by lbestock** [comment]
A screen shot of the one time I got the blank image container in experimental mode. As you can see, it turned all the other image containers blank, too.

**Comment by urapadmin** [comment]
if we can't repeat it I don't know how to proceed.

**Comment by lbestock** [comment]
Repeating my steps exactly did not repeat the problem. I cleared everything out. I downloaded again, and this time my second dayplan was not blank. The only difference is that now I have all my other tabs closed.

**Comment by urapadmin** [comment]
clueless. I am preparing a new version with a second take. But this just gets more and more screwed up.

**Comment by lbestock** [comment]
I can't get it to come back. Now I am like Luiza with no problems. If I didn't have that screenshot with experimental and a bunch of blank image containers I would think it is just me going crazy.

**Comment by urapadmin** [comment]
I take it that you always used the same unit etc. when testing today.

**Comment by lbestock** [comment]
I have used multiple units, but when I repeated my steps to try to see if the error was immediately after download I started in the same unit in which I had had the error.

**Comment by urapadmin** [comment]
this is getting out of hand, so next test in new ticket: #1246

**Comment by urapadmin** [comment]
#1246 is closed and our new ticket is now #1249.
I close this here because it is a mess.
